plugins/inputs/system/ps.go

```diff
@@ -154,7 +154,7 @@ func (s *SystemPS) DiskUsage(
 		du, err := s.PSDiskUsage(mountpoint)
 		if err != nil {
 			if s.Log != nil {
-				s.Log.Errorf("[SystemPS] => error getting disk usage (%q): %v", mountpoint, err)
```

Comment:
It makes sense for this to be a warning for 'permission denied' errors, but other (unexpected) causes probably are errors and should be logged at error level.

What do you think of adding a type assertion to the specific 'permission denied' error and logging at warn level only in that case?

Reply:
Here is a current example of the output:

```
2022-03-31T19:53:52Z E! [inputs.disk] [SystemPS] => error getting disk usage ("/var/lib/docker/btrfs"): permission denied
2022-03-31T19:53:52Z E! [inputs.disk] [SystemPS] => error getting disk usage ("/run/user/1000/doc"): operation not permitted
```

As we have noted with #10897, our users started seeing error messages and the messages caused a great deal of concern. They wished to resolve them thinking they may have done something wrong, a configuration value could resolve this, or stop seeing errors from showing up.

The second message above is from a `fuse.portal` filesystem mount. This does not prevent Telegraf from running and the disk plugin continues to gather data from other mounts.

The user experience of a warning seems more ideal than erroring about things that are not urgent to resolve (e.g. cannot connect to a service), especially when these were hidden in the past.

Thoughts?

Reply:
I'm fine with switching this to warn or even back to debug. I thought users would want to know what filesystems they're missing but the change in behavior seems to have been more trouble than it's worth.

Reply:
reverting all the way back to debug